File: unit-tests/parameterized_unit_tests/parameterized_unit_tests.jl
module ex
using ParameterTests
using Test

function palindrome_detector(s::String)::Bool
    s = replace(lowercase(s), " " => "")
    return s == s[end:-1:begin]
end

@paramtest "test_palindrome_detector" begin
    @given (input, expected) ∈ [("madam", true), ("false", false)]
    @test(palindrome_detector(input) == expected)
end

end

File: unit-tests/parameterized_unit_tests/parameterized_unit_tests.py
import pytest

def palindrome_detector(s: str):
    s = s.lower().replace(' ', '')
    return s == s[::-1]

@pytest.mark.parametrize("input,expected", [
    ("madam", True),
    ("false", False)])
def test_palindrome_detector(input, expected):
    assert palindrome_detector(input) == expected